File: LeetCode_Marathons/LCM2/Medium/Count_Primes/a.go
func countPrimes(n int) int {
    isNotPrimeStatuses := make([]bool, n)
    counter := 0
    for i := 2; i < n; i++ {
        if !isNotPrimeStatuses[i] {
            if math.Sqrt(float64(n)) >= float64(i) {
                for j := i * i; j < n; j += i {
                    isNotPrimeStatuses[j] = true
                }
            }
            counter++;
        }
    }
    return counter;
}

File: LeetCode_Marathons/LCM2/Medium/Count_Primes/a.java
class Solution {
    public int countPrimes(int n) {
        boolean[] isNotPrimeStatuses = new boolean[n + 1];
        int counter = 0;
        for (int i = 2; i < n; i++) {
            if (!isNotPrimeStatuses[i]) {
                if (Math.sqrt(n) >= i) {
                    for (int j = i * i; j < n; j += i) isNotPrimeStatuses[j] = true;
                }
                counter++;
            }
        }
        return counter;
    }
}